core/test/integration/database.ts

```diff
@@ -86,4 +86,50 @@ describe('Database', () => {
       expectedCollections[1],
     ]);
   });
+
+  it('should not drop any database or collection while removing all documents', async () => {
+    const expectedCollections = ['db-remove-doc-first', 'db-remove-doc-second'];
+
+    for (const collection of expectedCollections) {
+      await createCollection(database.db, collection);
+    }
+
+    await database.removeAllDocumentsIfCollectionExists(expectedCollections[0]);
+    await expect(listExistingCollections(database.db)).resolves.toStrictEqual(
+      expect.arrayContaining([expectedCollections[0], expectedCollections[1]]),
+    );
+  });
+
+  it('should remove all documents from collection without droping the db/collection', async () => {
+    const documents = [
+      {
+        value: 1,
+      },
+      {
+        name: 'two',
+        value: 2,
+      },
+      {
+        value: 3,
+        object: {
+          name: 'three',
+        },
+      },
+    ];
+    const collection = 'db-remove-docs';
+
+    await database.insertDocumentsIntoCollection(documents, collection);
+    const inserted = await database.db.collection(collection).find().toArray();
+    expect(inserted).toHaveLength(documents.length);
+
+    await database.removeAllDocumentsIfCollectionExists(collection);
+    const result = await database.db.collection(collection).find().toArray();
+    expect(result).toHaveLength(0);
+  });
```

Comment:
I think we can remove this test case as well, as it will be tested in `test/integration/index.ts`. While I know there are such examples on the repo, at the end I would like to reduce integration tests and do not duplicate them.